tokio/src/runtime/task/trace/mod.rs

```diff
@@ -317,20 +319,28 @@ cfg_rt_multi_thread! {
         drop(synced);
 
         // notify each task
-        let mut traces = vec![];
+        let mut tasks = vec![];
```

Comment:
There are some concurrent problems here. Did you notice that the tace all tasks implementation of current_thread is differenct from the implementation of multi_thread?

Because there are more than 1 worker threads in multi_thread, they could get the same tasks collection from OwnedTasks in the above trace_multi_thread method. Our repeated polling of the same Task in multi worker threads is not the key issue here. The key issue is that we are not allowed to poll the same task concurrently in multi worker threads.

We should find a way to avoid this concurrency issue.